candle-flash-attn/kernels/flash_api.cu

```diff
@@ -106,6 +109,19 @@ extern "C" void run_mha(
     params.cu_seqlens_q = cu_seqlens_q_ptr;
     params.cu_seqlens_k = cu_seqlens_k_ptr;
     params.p_ptr = nullptr; // used for `return_softmax`.
+    params.seqused_k = nullptr;
+
+    // Causal is the special case where window_size_right == 0 and window_size_left < 0.
```

Comment:
I would rather have this `is_causal` logic live on the rust side rather than here, this way it will be easier to tweak without having to recompile the cuda bits (and also it will benefit from being in rust and this would just be an ffi layer without any smartness into it).

Reply:
done